src/main/java/org/hydev/mcpm/client/interaction/ProgressBarBoundary.java

```diff
@@ -0,0 +1,50 @@
+package org.hydev.mcpm.client.interaction;
+
+import java.util.List;
+
+/**
+ * Terminal progress bar based on Xterm escape codes
+ *
+ * @author Azalea (https://github.com/hykilpikonna)
+ * @author Peter (https://github.com/MstrPikachu)
+ * @since 2022-09-27
+ */
+public interface ProgressBarBoundary extends AutoCloseable {
+    /**
+     * Append a progress bar at the end
+     *
+     * @param bar Row of the progress bar
+     * @return bar for fluent access
+     */
+    ProgressRowBoundary appendBar(ProgressRowBoundary bar);
+
+    /**
+     * Finish a progress bar
+     *
+     * @param bar Progress bar
+     */
+    void finishBar(ProgressRowBoundary bar);
+
+    /**
+     * Finalize and close the progress bar (print the final line)
+     */
+    @Override
+    void close();
```

Comment:
Why is this overridden and do we actually override this method with anything useful? Seems empty to me.

Reply:
My bad I for some reason added this here even though it gets inherited from AutoCloseable

Reply:
Yea for single-line progress bars, since they are using `\r` to reset the cursor to col 1 on each update, it needs to print a new line when the progress bar is done (i.e. closed). But I figured that multi-line progress bars don't need this... maybe we should remove it from the boundary?